src/main/scala/nl/biopet/tools/vcfstats/VcfStatsSpark.scala

```diff
@@ -0,0 +1,402 @@
+
+      val contigFutures = if (!cmdArgs.value.notWriteContigStats) {
+        Some(generalContig.foreachAsync {
+          case (contig, stats) =>
+            val dir = contigDir(cmdArgs.value.outputDir, contig)
+            dir.mkdirs()
+            stats.writeToTsv(new File(dir, "general.tsv"))
+        })
+      } else None
+      vcfRecords.sparkContext.clearJobGroup()
+      (contigFutures, Some(generalContig))
+    }
+  }
+
+  /** This will combine contigs stats and write the output file */
+  def totalGeneralStats(generalContig: List[RDD[(String, GeneralStats)]],
+                        cmdArgs: Broadcast[Args]): Option[Future[Unit]] = {
+    if (generalContig.nonEmpty) {
+      val sc = generalContig.head.context
+      sc.setJobGroup("Total General stats", "Total General stats")
+      val emptyRdd = sc.parallelize(
+        List("total" -> new GeneralStats()))
+      Some(sc.union(emptyRdd :: generalContig)
+        .map("total" -> _._2)
+        .reduceByKey(_ += _)
+        .foreachAsync(
+          _._2.writeToTsv(new File(cmdArgs.value.outputDir, "general.tsv"))))
+    } else None
+  }
+
+  /** This will generate contigs stats and write this if this is not disabled */
+  def contigGenotypeStats(vcfRecords: RDD[VariantContext],
+                          header: Broadcast[VCFHeader],
+                          cmdArgs: Broadcast[Args],
+                          regions: Broadcast[List[BedRecord]],
+                          prefixMessage: String): (Option[FutureAction[Unit]], Option[RDD[(String, GenotypeStats)]]) = {
+    if (cmdArgs.value.skipGeneral) (None, None)
+    else {
+      vcfRecords.sparkContext.setJobGroup(s"$prefixMessage Genotype stats", s"$prefixMessage Genotype stats")
+      val genotypeContig = spark.vcf.genotypeStats(vcfRecords, header, regions).cache()
+      val contigFutures = if (!cmdArgs.value.notWriteContigStats) {
+        Some(genotypeContig.foreachAsync {
+          case (contig, stats) =>
+            val dir = contigDir(cmdArgs.value.outputDir, contig)
+            dir.mkdirs()
+            stats.writeToTsv(new File(dir, "genotype.tsv"))
+        })
+      } else None
+      vcfRecords.sparkContext.clearJobGroup()
+      (contigFutures, Some(genotypeContig))
+    }
+  }
+
+  /** This will combine contigs stats and write the output file */
+  def totalGenotypeStats(contigs: List[RDD[(String, GenotypeStats)]],
+                         cmdArgs: Broadcast[Args],
+                         header: Broadcast[VCFHeader]): Option[Future[Unit]] = {
+    if (contigs.nonEmpty) {
+      val sc = contigs.head.context
+      sc.setJobGroup("Total Genotype stats", "Total Genotype stats")
+      val emptyRdd = sc.parallelize(
+        List("total" -> new GenotypeStats(header.value)))
+      Some(sc.union(emptyRdd :: contigs)
+        .map("total" -> _._2)
+        .reduceByKey(_ += _)
+        .foreachAsync(
+          _._2.writeToTsv(new File(cmdArgs.value.outputDir, "genotype.tsv"))))
+    } else None
```

Comment:
To illustrate the comment  about the ._2. Something  is written to a tsv file here. But not very clear what exactly. A total consisting of `._2` reduced by `_+=_` and after that `foreachAsync(_._2`.

Reply:
See comment above